### PR TITLE
Paper: Python for Global Applications

### PR DESCRIPTION
**Pull request by annahaensch**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

*Editor*: @stargaser

*Reviewers*: @stargaser

### CODE REVIEW

**Comment by scoobies** [comment]
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

**Comment by deniederhut** [comment]
@scoobies mark paper

**Comment by deniederhut** [comment]
@scoobies build status

**Comment by scoobies** [comment]
{"annahaensch-2022"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpgd75cojs'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: anna_haensch\n*** Warning: /tmp/tmpgd75cojs/publisher/../output/anna_haensch/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/annahaensch-2022.pdf", "build_status"=>"success", "build_timestamp"=>"23/05 16:52"}, "status"=>"success"}}

**Comment by deniederhut** [comment]
@scoobies mark needs review

**Comment by deniederhut** [comment]
@scoobies assign @stargaser as editor

**Comment by scoobies** [comment]
Assigned! @stargaser is now the editor

**Comment by deniederhut** [comment]
Hi! We've moved the deadline for first reviews back one week, to June 16th, to give us a little more time to find reviewers for all the papers this year. You can see the updated full timeline at https://github.com/scipy-conference/scipy_proceedings#timeline-for-2022.

**Comment by deniederhut** [comment]
@scoobies add @cmahapatra97 to reviewers

**Comment by scoobies** [comment]
@cmahapatra97 added to the reviewers list!

**Comment by stargaser** [comment]
Hi @cmahapatra97, have you had a chance to review this paper? The nominal deadline is coming up tomorrow.

**Comment by deniederhut** [comment]
@scoobies remove @cmahapatra97 from reviewers

**Comment by scoobies** [comment]
@cmahapatra97 removed from the reviewers list!

**Comment by deniederhut** [comment]
@scoobies add @stargaser to reviewers

**Comment by scoobies** [comment]
@stargaser added to the reviewers list!

File: papers/anna_haensch/paper.rst [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+Connecting the Technical and Social
+******************************************
+
+To understand the impact of data science on the world (and the potential policy implications of such impact), it helps to have hands-on practice with data science. Conversely, to effectively and ethically practice data science, it is important to understand how data science lives in the world. Thus, the "hard" skills of coding, wrangling data, visualizing, and modeling is best taught intertwined with a robust study of ways in which data science is used and misused.
```
Comment (stargaser) [inline papers/anna_haensch/paper.rst]: hard skills are best taught

File: papers/anna_haensch/paper.rst [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+We contend that the value of supported programming time is two-fold. First, it helps beginning programmers learn more quickly. While learning to code necessarily involves challenges, students new to a language can sometimes struggle for an unproductively long time on things like simple syntax issues. When students have help available, they can move forward from minor issues faster and move more efficiently into building a meaningful understanding. Secondly, supported programming time helps students to understand that they are not alone in the challenges they are facing in learning to program.  They can see other students learning and facing similar challenges, can have the empowering experience of helping each other out, and when asking for help can notice that even their instructors sometimes rely on resources like StackOverflow. An unforeseen benefit we believe co-teaching had was to give us as instructors the opportunity to consult with each other during class time and share different approaches. These instrutor interactions modeled for students how even as experienced practitioners of data science, we too were constantly learning.
```
Comment (stargaser) [inline papers/anna_haensch/paper.rst]: * Web link for Stack Overflow
* "These instructor interactions"

File: papers/anna_haensch/paper.rst [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+Lastly, a small but (we thought) important aspect of our setup was teaching students to set up a computing environment on their own laptops, with Python, conda, and JupyterLab. Using the command line and moving from an environment like Google Colab to one’s own computer can both present significant barriers, but doing so successfully can be an important part of helping students feel like ‘real’ programmers. We devoted an entire class period to helping students with installation and setup on their own computers.
```
Comment (stargaser) [inline papers/anna_haensch/paper.rst]: Reference for Jupyterlab / Jupyter, and conda if available

File: papers/anna_haensch/paper.rst [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+We can point to several key aspects of the course that made it successful.  Primary among them was the fact that the majority of class time was spent in supported programming.  This means that students were able to ask their instructors or peers as soon as questions arose.  Novice programmers who aren't part of a formal computer science program often don't have immediate access to the resources necessary to get "unstuck." for the novice practitioner, even learning *how* to google technical terms can be a challenge.  This sort of immediate debugging and feedback helped students remain confident and optimistic about their projects.  This was made all the more effective since we were co-teaching the course and had double the resources to troubleshoot. Co-teaching also had the unforeseen benefit of making our classroom a place where the growth mindset was actively modeled and nurtured: where one instructor wasn't able to answer a question, the other instructor often could. Finally, it was precisely the motivation of learning data science in context that allowed students to maintain a sense of ownership over their work and build connections between their other courses.
```
Comment (stargaser) [inline papers/anna_haensch/paper.rst]: For the novice programmer

**Review by stargaser — COMMENTED** [review]
This paper is a very accessible account of the authors' experience co-teaching a data science course for students in international relations and diplomacy. The paper is well-written and I don't have suggestions for improvement beyond a few in-line comments.

**Comment by stargaser** [comment]
@scoobies mark pending comment

**Comment by annahaensch** [comment]
Thank you @stargaser for the helpful comments and suggestions. All of your recommended changes have been implemented and the discussions have been resolved.

**Comment by stargaser** [comment]
@scoobies mark ready

**Pull request merged**